pkg/cli/options.go

```diff
@@ -64,11 +65,25 @@ type Executable interface {
 func ExecOptions(c *Config, opts Executable) func(cmd *cobra.Command, args []string) error {
 	return func(cmd *cobra.Command, args []string) error {
 		ctx := WithCommand(context.Background(), cmd)
+		if opts.IsDryRun() {
+			// reserve Stdout for resources, redirect normal stdout to stderr
+			ctx = withStdout(ctx, c.Stdout)
+			c.Stdout = c.Stderr
+		}
 		return opts.Exec(ctx, c)
 	}
 }
 
+type CommonOptions struct {
```

Comment:
I find this introduction a bit dangerous and not really necessary. Go tends to use very targeted interfaces, that can be composed on demand.

My fear here is that next thing you know, `CommonOptions` will become a kitchen sink.

To put it another way, why not introduce something dedicated to DryRun and DryRun only?

Reply:
Also, would be better to dynamically check whether the `Executable opts` conform to "DryRunnable" IMO.

Reply:
done